docs/oauth2client-deprecation.rst

```diff
@@ -0,0 +1,62 @@
+
+.. _httplib2: https://github.com/httplib2/httplib2
+
+We originally planned to address these issues within ``oauth2client``, however, we determined that the number of breaking changes needed would be absolutely untenable for downstream users. It would essentially involve our users having to rewrite significant portions of their code if they needed to upgrade (either directly or indirectly through a dependency). Instead, we’ve chosen to create a new replacement library that can live side-by-side with ``oauth2client`` and allow users to migrate gradually. We believe that this was the least painful option.
+
+Replacement
+-----------
+
+The long-term replacement for ``oauth2client`` is this library, ``google-auth``. This library addresses the major issues with oauthclient:
+
+#. Clear ownership: google-auth is owned by the teams that maintain the `Cloud Client Libraries`_, `gRPC`_, and the `Code Samples for Google Cloud Platform`_.
+#. Thought-out design: using the lessons learned from ``oauth2client``, we have designed a better module and class hierarchy. The ``v1.0.0`` release of this library should provide long-term API stability.
+#. Modern, secure, and extensible transports: ``google-auth`` supports `urllib3`_, `requests`_, `gRPC`_, and has `legacy support for httplib2`_ to help clients migration. It is transport agnostic and has explicit support for adding new transports.
+#. Clear purpose and goals: ``google-auth`` is explicitly focused on Google-specific authentication, especially the server-to-server use case.
+ 
+Because we reduced the scope of the library, there are several features in ``oauth2client`` we intentionally are not supporting in the ``v1.0.0`` release of ``google-auth``. This does not mean we are not interested in supporting these features, we just didn't feel they should be part of the initial API. As downstream users ask for these features we will determine the best way to serve those use cases without allowing the library to become a dumping ground.
+ 
+The unsupported features are:
+
+#. Support for obtaining user credentials. While this library has support for using user credentials, there are no provisions in the core library for doing the three-party OAuth 2.0 flow to obtain authorization from a user. Instead, we are opting to provide a separate package that does integration with `oauthlib`_, `google-auth-oauthlib`_. When that library has a stable API, we will consider its inclusion into the core library.
+#. Support for storing credentials. The only credentials type that needs to be stored are user credentials. We have a `discussion thread`_ on what level of support we should do. It’s very likely we’ll choose to provide an abstract interface for this and leave it up to application to provide storage implementation specific to their use case. It's also very likely that we will also incubate this functionality in the `google-auth-oauthlib`_ library before including it in the core library.
```

Comment:
The single quotes in "It's" and "we'll" are a unicode right quote instead of an ASCII single quote (How dare you!)